Building a POM without dependencies results into an empty query resulting in a Bad Request

```txt
➜  maven-archetypes git:(master) mvn com.giovds:outdated-maven-plugin:1.0.0:check
[INFO] Unable to find the root directory. Create a .mvn directory in the root directory or add the root="true" attribute on the root project's model to identify it.
[INFO] Scanning for projects...
[INFO] --------------------------------------------------------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO]
[INFO] Apache Maven Archetypes                                                                                              [pom]
[INFO] Maven Archetype Archetype                                                                                [maven-archetype]
[INFO] Maven Simple J2EE Archetype                                                                              [maven-archetype]
[INFO] Maven Plugin Archetype                                                                                   [maven-archetype]
[INFO] Maven Plugin Site Archetype                                                                              [maven-archetype]
[INFO] Maven Portlet Archetype                                                                                  [maven-archetype]
[INFO] Maven Quickstart Archetype                                                                               [maven-archetype]
[INFO] Maven Simple Project Archetype                                                                           [maven-archetype]
[INFO] Maven Site Archetype                                                                                     [maven-archetype]
[INFO] Maven Archetype for Simple Site                                                                          [maven-archetype]
[INFO] Maven Site Skin Archetype                                                                                [maven-archetype]
[INFO] Maven Webapp Archetype                                                                                   [maven-archetype]
[INFO]
[INFO] ---------------------------------< org.apache.maven.archetypes:maven-archetype-bundles >----------------------------------
[INFO] Building Apache Maven Archetypes 1.5-SNAPSHOT                                                                       [1/12]
[INFO]   from pom.xml
[INFO] ---------------------------------------------------------[ pom ]----------------------------------------------------------
[INFO]
[INFO] --- outdated:1.0.0:check (default-cli) @ maven-archetype-bundles ---
[INFO] --------------------------------------------------------------------------------------------------------------------------
[INFO] Reactor Summary for Apache Maven Archetypes 1.5-SNAPSHOT:
[INFO]
[INFO] Apache Maven Archetypes .............................................................................. FAILURE [  0.825 s]
[INFO] Maven Archetype Archetype ............................................................................ SKIPPED
[INFO] Maven Simple J2EE Archetype .......................................................................... SKIPPED
[INFO] Maven Plugin Archetype ............................................................................... SKIPPED
[INFO] Maven Plugin Site Archetype .......................................................................... SKIPPED
[INFO] Maven Portlet Archetype .............................................................................. SKIPPED
[INFO] Maven Quickstart Archetype ........................................................................... SKIPPED
[INFO] Maven Simple Project Archetype ....................................................................... SKIPPED
[INFO] Maven Site Archetype ................................................................................. SKIPPED
[INFO] Maven Archetype for Simple Site ...................................................................... SKIPPED
[INFO] Maven Site Skin Archetype ............................................................................ SKIPPED
[INFO] Maven Webapp Archetype ............................................................................... SKIPPED
[INFO] --------------------------------------------------------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] --------------------------------------------------------------------------------------------------------------------------
[INFO] Total time:  0.969 s
[INFO] Finished at: 2024-07-15T21:55:56+02:00
[INFO] --------------------------------------------------------------------------------------------------------------------------
[ERROR] Failed to execute goal com.giovds:outdated-maven-plugin:1.0.0:check (default-cli) on project maven-archetype-bundles: Failed to connect.: com.fasterxml.jackson.core.JsonParseException: Unrecognized token 'Solr': was expecting (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
[ERROR]  at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 6]
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the '-e' switch
[ERROR] Re-run Maven using the '-X' switch to enable verbose output
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException```

--- a/src/main/java/com/giovds/OutdatedMavenPluginMojo.java
+++ b/src/main/java/com/giovds/OutdatedMavenPluginMojo.java
@@ -52,6 +52,12 @@ public class OutdatedMavenPluginMojo extends AbstractMojo {
     @Override
     public void execute() throws MojoExecutionException, MojoFailureException {
         final String queryForAllDependencies = DependenciesToQueryMapper.map(project.getDependencies());
+
+        if (queryForAllDependencies.isBlank()) {
+            // When building a POM without any dependencies there will be nothing to query.
+            return;
+        }
+
         final Set<QueryClient.FoundDependency> result = client.search(queryForAllDependencies);
 
         final Collection<QueryClient.FoundDependency> outdatedDependencies = new ArrayList<>();

--- a/src/test/java/com/giovds/OutdatedMavenPluginMojoTest.java
+++ b/src/test/java/com/giovds/OutdatedMavenPluginMojoTest.java
@@ -9,13 +9,17 @@ import org.mockito.InjectMocks;
 import org.mockito.Mock;
 
 import java.time.LocalDate;
+import java.util.Collections;
 import java.util.List;
 import java.util.Set;
 
 import static org.assertj.core.api.AssertionsForClassTypes.assertThatCode;
 import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
 import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.anyString;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 class OutdatedMavenPluginMojoTest {
@@ -38,11 +42,21 @@ class OutdatedMavenPluginMojoTest {
     }
 
     @Test
-    void should_now_throw_exception_by_default_when_outdatedDependencies() throws Exception {
+    void should_not_throw_exception_by_default_when_outdatedDependencies() throws Exception {
         final MavenProject projectWithDependency = createProjectWithDependencyOfAge(LocalDate.now().minusYears(10));
         mojo.setProject(projectWithDependency);
 
         assertThatCode(() -> mojo.execute()).doesNotThrowAnyException();
+    }
+
+    @Test
+    void should_not_throw_exception_when_no_dependencies() throws MojoExecutionException {
+        final MavenProject project = new MavenProject();
+        project.setDependencies(Collections.emptyList());
+        mojo.setProject(project);
+
+        assertThatCode(() -> mojo.execute()).doesNotThrowAnyException();
+        verify(client, never()).search(anyString());
     }
 
     private MavenProject createProjectWithDependencyOfAge(final LocalDate timestamp) throws MojoExecutionException {

